spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -69,7 +88,7 @@ public void setResourceLoader(ResourceLoader resourceLoader) {
 
 	private GoogleStorageProtocolResolverSettings getSettings() {
 		try {
-			return this.beanFactory.getBean(GoogleStorageProtocolResolverSettings.class);
+			return getGoogleStorageProtocolResolverSettings();
```

Comment:
Why not inlining those getters here?
e.g.,
```
if (this.googleStorageProtocolResolverSettings == null) {
this.googleStorageProtocolResolverSettings = this.beanFactory.getBean(GoogleStorageProtocolResolverSettings.class);
}
return this.googleStorageProtocolResolverSettings;
```

Even better, do it in `postProcessBeanFactory()`, after `this.beanFactory = beanFactory`.

Reply:
I had that originally , but it appears things are not all ready to be used by that point. Specifically the GoogleProtocolResolverSettings isn't correct to be retrieved from the beanFactory at that point and unit tests fail.

Reply:
We can't do that in the `postProcessBeanFactory()`. That is going to be *evil* called `early BeanFactory` access.
That's why we resolve these bean lazy-load manner.